core/src/main/java/org/apache/iceberg/actions/RewriteDeleteStrategy.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+import java.util.Map;
+import org.apache.iceberg.DeleteFile;
+import org.apache.iceberg.Table;
+
+public interface RewriteDeleteStrategy {
+
```

Comment:
I think we also need the methods similar to `RewriteStrategy` such as `name`, `validOptions` and `options`

Reply:
I didn't add `validOptions` because we haven't finalized what options are valid and it can be added late. But I'm fine to add them now. 

For `name`, It looks like the class name already includes the strategy name.  Do we still need it?

Reply:
`name` is mostly used for logging purpose. I agree it is already in the class name, but for consistency with `RewriteStrategy` I think we can still have it here.